# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1680722105 -7200
# Parent  58c0dfead17ac02c88874d3bcb0bc4c772b422e1
Feature: When multiple message are selected, current message is shown.

Contains part of one hunk of Mats Palmgren's patch from https://bugzilla.mozilla.org/show_bug.cgi?id=364896

diff --git a/mailnews/base/src/nsMsgDBView.cpp b/mailnews/base/src/nsMsgDBView.cpp
--- a/mailnews/base/src/nsMsgDBView.cpp
+++ b/mailnews/base/src/nsMsgDBView.cpp
@@ -1220,30 +1220,29 @@ nsMsgDBView::SelectionChangedXPCOM() {
       mSummarizeFailed = true;
     }
   }
 
   bool summaryStateChanged = selectionSummarized != mSelectionSummarized;
   mSelectionSummarized = selectionSummarized;
 
   // If only one item is selected then we want to display a message.
-  if (mTreeSelection && selection.Length() == 1 && !selectionSummarized) {
-    int32_t startRange;
-    int32_t endRange;
-    nsresult rv = mTreeSelection->GetRangeAt(0, &startRange, &endRange);
+  // If we're not summarized, we display the current message.
+  if (mTreeSelection && selection.Length() >= 1 && !selectionSummarized) {
+    int32_t currentIndex;
+    nsresult rv = mTreeSelection->GetCurrentIndex(&currentIndex);
     // Tree doesn't care if we failed.
     NS_ENSURE_SUCCESS(rv, NS_OK);
 
-    if (startRange >= 0 && startRange == endRange &&
-        uint32_t(startRange) < GetSize()) {
+    if (currentIndex >= 0 && uint32_t(currentIndex) < GetSize()) {
       if (!mRemovingRow) {
         if (!mSuppressMsgDisplay)
-          LoadMessageByViewIndex(startRange);
+          LoadMessageByViewIndex(currentIndex);
         else
-          UpdateDisplayMessage(startRange);
+          UpdateDisplayMessage(currentIndex);
       }
     } else {
       // Selection seems bogus, so set to 0.
       selection.Clear();
     }
   } else {
     // If we have zero or multiple items selected, we shouldn't be displaying
     // any message.
